src/openforms/tests/e2e/test_input_validation.py

```diff
@@ -292,3 +317,209 @@ def test_max_date_fixed_value(self):
+            "validate": {"required": True},
+        }
+
+        self.assertValidationIsAligned(
+            component,
+            ui_input="",
+            api_value=False,
+            expected_ui_error="Het verplichte veld Required checkbox is niet ingevuld.",
+        )
+
+
+class SingleCurrencyTests(ValidationsTestCase):
+    def test_required_field(self):
+        component: Component = {
+            "type": "currency",
+            "key": "requiredCurrency",
+            "label": "Required currency",
+            "validate": {"required": True},
+        }
+
+        self.assertValidationIsAligned(
+            component,
+            ui_input="",
+            expected_ui_error="Het verplichte veld Required currency is niet ingevuld.",
+        )
+
+    def test_min_value(self):
+        component: Component = {
+            "type": "currency",
+            "key": "requiredCurrency",
+            "label": "Required currency",
+            "validate": {"min": 10.7},
+        }
+
+        self.assertValidationIsAligned(
+            component,
+            ui_input=2,
+            expected_ui_error="De waarde moet 10.7 of groter zijn.",
+        )
+
+    def test_max_value(self):
+        component: Component = {
+            "type": "currency",
+            "key": "requiredCurrency",
+            "label": "Required currency",
+            "validate": {"max": 15},
+        }
+
+        self.assertValidationIsAligned(
+            component,
+            ui_input=50,
+            expected_ui_error="De waarde moet 15 of kleiner zijn.",
+        )
+
+
+class SingleMapTests(ValidationsTestCase):
+    async def apply_ui_input(self, page: Page, label: str, ui_input: str | int | float):
+        page.wait_for_selector(
+            ".openforms-leaflet-map, [aria-label='Required map']", state="visible"
+        )
+
+    def test_required_field(self):
+        component: Component = {
+            "type": "map",
+            "key": "requiredMap",
+            "label": "Required map",
+            "validate": {"required": True},
+        }
+
+        self.assertValidationIsAligned(
+            component,
+            ui_input="",
+            expected_ui_error="Het verplichte veld Required map is niet ingevuld.",
+        )
+
+
+class SinglePostcodeTests(ValidationsTestCase):
+    def test_required_field(self):
+        component: Component = {
+            "type": "postcode",
+            "key": "requiredPostcode",
+            "label": "Required postcode",
+            "validate": {"required": True},
+        }
+
+        self.assertValidationIsAligned(
+            component,
+            ui_input="",
+            expected_ui_error="Het verplichte veld Required postcode is niet ingevuld.",
+        )
+
+
+class SingleSignatureTests(ValidationsTestCase):
+
+    async def apply_ui_input(self, page: Page, label: str, ui_input: str | int | float):
+        page.wait_for_selector("[aria-label='Required signature']", state="visible")
```

Comment:
same remark about hardcoed label here :)

Reply:
This test was failing when I was using `await` so I assumed that it is not needed, that's why I removed it from above as well (the one above though is passing in both ways).

Reply:
I checked out your branch locally and am running the tests and as expected, there is a warning :)

```
src/openforms/tests/e2e/test_input_validation.py:427: RuntimeWarning: coroutine 'Page.wait_for_selector' was never awaited
  page.wait_for_selector(f"[aria-label='{label}']", state="visible")
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
```